src/py/flwr/server/run_serverapp.py

```diff
@@ -219,15 +219,15 @@ def _parse_args_run_server_app() -> argparse.ArgumentParser:
     )
 
     parser.add_argument(
-        "server-app",
+        "app_dir",
```

Comment:
```suggestion
        "app-dir",
```

Reply:
<img width="646" alt="Screenshot 2024-08-03 at 21 10 32" src="https://github.com/user-attachments/assets/14d4fdcd-8de4-44c0-ba45-325b61a0a712">
I think the idea is to align with `flwr run`. So I use `_` instead of `-` here.

Reply:
Second thought, we use `--flwr-dir` instead of `--flwr_dir`, which is different from the case here. What naming convention we should preserve then? @danieljanes @jafermarq ?

Reply:
I agree, let's use `app-dir`. We use `-` instead of `_` in all CLI args. In `flwr run`, the reason it says `app_dir` is b/c it's a positional arg. I believe there's not way to display it as `app-dir` there, but if there is, we should change it there as well.

One alternative would be to shorten it to `app` everywhere @panh99 @jafermarq @charlesbvll

Reply:
Sounds good. We can provide that clarification via the documentation and docstrings. With the next release, pointing at the `ClientApp` will be a thing of the past. This makes me optimistic that not many users will be confused.

Reply:
Cool. I will rename it to `app` in this PR then. Any thought on this @jafermarq @charlesbvll ?

Reply:
Sounds good to me!